src/main/java/com/gregtechceu/gtceu/common/machine/multiblock/electric/CentralMonitorMachine.java

```diff
@@ -0,0 +1,148 @@
+package com.gregtechceu.gtceu.common.machine.multiblock.electric;
+
+import com.gregtechceu.gtceu.api.machine.IMachineBlockEntity;
+import com.gregtechceu.gtceu.api.machine.MachineDefinition;
+import com.gregtechceu.gtceu.api.machine.multiblock.PartAbility;
+import com.gregtechceu.gtceu.api.machine.multiblock.WorkableElectricMultiblockMachine;
+import com.gregtechceu.gtceu.api.machine.multiblock.WorkableMultiblockMachine;
+import com.gregtechceu.gtceu.api.pattern.BlockPattern;
+import com.gregtechceu.gtceu.api.pattern.FactoryBlockPattern;
+import com.gregtechceu.gtceu.api.pattern.Predicates;
+import com.gregtechceu.gtceu.api.pattern.TraceabilityPredicate;
+import com.gregtechceu.gtceu.api.pattern.util.RelativeDirection;
+import com.gregtechceu.gtceu.common.data.GTBlocks;
+import com.gregtechceu.gtceu.common.data.GTMachines;
+import com.gregtechceu.gtceu.common.data.machines.GTResearchMachines;
+
+import com.lowdragmc.lowdraglib.syncdata.field.ManagedFieldHolder;
+
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class CentralMonitorMachine extends WorkableElectricMultiblockMachine {
+
+    protected static final ManagedFieldHolder MANAGED_FIELD_HOLDER = new ManagedFieldHolder(CentralMonitorMachine.class,
+            WorkableMultiblockMachine.MANAGED_FIELD_HOLDER);
+
+    private static final Block[] VALID_BLOCKS = new Block[] {
+            GTBlocks.CASING_ALUMINIUM_FROSTPROOF.get()
+    };
+    private static final Set<MachineDefinition[]> VALID_MACHINES = Set.of(
+            GTMachines.HULL,
+            GTMachines.ENERGY_INPUT_HATCH,
+            GTMachines.ENERGY_INPUT_HATCH_4A,
+            GTMachines.ENERGY_INPUT_HATCH_16A,
+            new MachineDefinition[] {
+                    GTResearchMachines.DATA_ACCESS_HATCH,
+                    GTResearchMachines.ADVANCED_DATA_ACCESS_HATCH,
+                    GTResearchMachines.CREATIVE_DATA_ACCESS_HATCH
+            });
+    public static final TraceabilityPredicate BLOCK_PREDICATE = Predicates.abilities(PartAbility.INPUT_ENERGY)
+            .setExactLimit(1)
+            .or(Predicates.abilities(PartAbility.DATA_ACCESS).setMaxGlobalLimited(1))
+            .or(Predicates.machines(GTMachines.HULL))
+            .or(Predicates.blocks(VALID_BLOCKS));
```

Comment:
this array can be inlined into the predicate.

I'm pretty sure making the predicate a static field won't work because the block registry won't be loaded when this class is initialized, so the predicate should just be part of 

```suggestion
    public static final TraceabilityPredicate BLOCK_PREDICATE = Predicates.abilities(PartAbility.INPUT_ENERGY)
            .setExactLimit(1)
            .or(Predicates.abilities(PartAbility.DATA_ACCESS).setMaxGlobalLimited(1))
            .or(Predicates.machines(GTMachines.HULL))
            .or(Predicates.blocks(GTBlocks.CASING_ALUMINIUM_FROSTPROOF.get()));
```